internal/kubernetes/client.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2017-present SIGHUP s.r.l All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package kubernetes
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/sighupio/furyctl/internal/tool/kubectl"
+	execx "github.com/sighupio/furyctl/internal/x/exec"
+	"github.com/sighupio/furyctl/internal/x/slices"
+)
+
+var logRegex = regexp.MustCompile(`'(.*?)'`)
+
+type Ingress struct {
+	Name string
+	Host []string
+}
+
+type Client struct {
+	kubeRunner *kubectl.Runner
+}
+
+func NewClient(
+	binPath,
+	workDir,
+	kubeconfig string,
+	serverSide,
+	skipNotFound,
+	clientVersion bool,
+	executor execx.Executor,
+) *Client {
+	return &Client{
+		kubeRunner: kubectl.NewRunner(
+			executor,
+			kubectl.Paths{
+				Kubectl:    binPath,
+				WorkDir:    workDir,
+				Kubeconfig: kubeconfig,
+			},
+			serverSide,
+			skipNotFound,
+			clientVersion,
+		),
+	}
+}
+
+func (c *Client) GetIngresses() ([]Ingress, error) {
+	var result []Ingress
+
+	log, err := c.kubeRunner.Get("all", "ingress", "-o",
+		"jsonpath='[{range .items[*]}{\"{\"}\"Name\": \"{.metadata.name}\", "+
+			"\"Host\": [{range .spec.rules[*]}\"{.host}\",{end}]{\"}\"},{end}]'")
+	if err != nil {
+		return result, fmt.Errorf("error while reading resources from cluster: %w", err)
+	}
+
+	logStringIndex := logRegex.FindStringIndex(log)
```

Comment:
I did a test with a local cluster with some ingresses, I ran the following command:
```bash
kubectl get ingress -A -o jsonpath='[{range .items[*]}{"{"}"Name": "{.metadata.name}", Host": [{range .spec.rules[*]}"{.host}",{end}]{"}"},{end}]'
```

and the regex is not matching anything... 🤔  see: https://regex101.com/r/51LS6d/1

so I expect that `FindStringIndex(log)` will return `nil` when it should not in this case.

I mean, I guess you tested this and works, so probably I am missing something, for example, I don't see where the `'` that the regex searches would come from. Is it something that the `kubeRunner.Get()` adds?

Reply:
yes, ```jsonpath='[{range .items[*]}{\"{\"}\"Name\": \"{.metadata.name}\", \"Host\": [{range .spec.rules[*]}\"{.host}\",{end}]{\"}\"},{end}]'"```kubeRunner encloses the output of the command between '' so that any output kubectl returns that is not from jsonpath (example "No resource found") won't be parsed by the regex.